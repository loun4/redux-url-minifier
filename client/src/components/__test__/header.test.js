
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import RouterProvider from '../../containers/router-provider';
import Header from '../header';

const deauthenticate = jest.fn();

const buildSubject = isAuthenticated => (
  <RouterProvider>
    <Header isAuthenticated={isAuthenticated} deauthenticate={deauthenticate} />
  </RouterProvider>
);

test('Render header before authentication', () => {
  const component = renderer.create(buildSubject(false));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render header after authentication', () => {
  const component = renderer.create(buildSubject(true));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('call deauthenticate', () => {
  const component = mount(buildSubject(true));
  component.find('.sign').simulate('click');
  expect(deauthenticate).toHaveBeenCalled();
});
