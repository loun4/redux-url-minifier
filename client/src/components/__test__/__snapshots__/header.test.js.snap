// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render header after authentication 1`] = `
<div
  className="ui inverted menu"
>
  <a
    className="header item active"
    href="/"
    onClick={[Function]}
  >
    URL shortener
  </a>
  <div
    className="right menu"
  >
    <a
      className="item"
      href="/admin"
      onClick={[Function]}
    >
      Admin
    </a>
    <button
      className="item"
      onClick={[Function]}
    >
      <i
        className="sign out alternate icon"
      />
      Sign out
    </button>
  </div>
</div>
`;

exports[`Render header before authentication 1`] = `
<div
  className="ui inverted menu"
>
  <a
    className="header item active"
    href="/"
    onClick={[Function]}
  >
    URL shortener
  </a>
  <div
    className="right menu"
  >
    <a
      className="item"
      href="/admin"
      onClick={[Function]}
    >
      Admin
    </a>
  </div>
</div>
`;
